### PR TITLE
Prepare `flutter_gallery` for `use_super_parameters`

### PR DESCRIPTION
**Pull request by goderbauer**
https://github.com/flutter/flutter/issues/100575

Once all code in the repo has been migrated, we can turn on the `use_super_parameters` lint.

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Review by Piinks — APPROVED** [review]
LGTM with musings. :)

File: dev/integration_tests/flutter_gallery/lib/demo/shrine/backdrop.dart [hunk 1]
```diff
@@ -14,10 +14,9 @@ const double _kPeakVelocityProgress = 0.379146;
 class _TappableWhileStatusIs extends StatefulWidget {
   const _TappableWhileStatusIs(
       this.status, {
-        Key? key,
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply dev/integration_tests/flutter_gallery/lib/demo/shrine/backdrop.dart:17]: These are private widgets and the `key` parameter is unused. The `unused_element` lint would trigger here if `super.key` were to be used.

[the remark this answers is not in the document]
Comment (Piinks) [reply dev/integration_tests/flutter_gallery/lib/demo/shrine/backdrop.dart:17]: So clean! Lints are the future. 🤖

**Pull request merged**